docs/basics/101-145-hooks.rst

```diff
@@ -0,0 +1,162 @@
+.. _hooks:
+
+DataLad's result hooks
+^^^^^^^^^^^^^^^^^^^^^^
+
+If you are particularly keen on automating tasks in your datasets, you may be
+interested in running DataLad commands automatically as soon
+as previous commands are executed and resulted in particular outcomes or states.
+For example, you may want to automatically :command:`unlock` all dataset contents
+right after an installation in one go. Therefore, you would like to automatically
+run the :command:`datalad unlock .` command right after the :command:`datalad install`
+command, *but only* if the previous :command:`install` command was successful.
+
+Such automation allows for flexible and yet automatic responses to the results
+of DataLad commands, and can be done with DataLad's *result hooks*.
+Generally speaking, `hooks <https://en.wikipedia.org/wiki/Hooking>`__ intercept
+function calls or events and allow to extend the functionality of a program.
+DataLad's result hooks are calls to other DataLad commands after the command
+resulted in a specified result -- such as a successful install.
+
+To understand how hooks can be used and defined, we have to briefly mention
+DataLad's *command result evaluations*. Whenever a DataLad
+command is executed, an internal evaluation generates a *report* on the status
+and result of the command. Internally, this is useful for final result
+rendering, error detection, and logging. However, by using hooks, you can
+utilize these evaluations for your own purposes and "hook" in more commands
+whenever an evaluation fulfills your criteria.
+
+To be able to specify matching criteria, you need to be aware of the potential
+criteria you can match against. The evaluation report is a dictionary with
+``key:value`` pairs. The following table provides an overview on some of the
+available keys and their possible values:
+
+.. list-table::
+   :widths: 50 100
+   :header-rows: 1
+
+   * - Key name
+     - Values
+   * - ``action``
+     - ``get``, ``install``, ``drop``, ``status``, ... (any command's name)
+   * - ``type``
+     - ``file``, ``dataset``, ``symlink``, ``directory``
+   * - ``status``
+     - ``ok``, ``notneeded``, ``impossible``, ``error``
+   * - ``path``
+     - The path the previous command operated on
```

Comment:
This is a good selection. The set is, however, not limited at all. For example, any third party extension could invent completely unforeseeable keys.